### PR TITLE
Allow copy paste element grid explore

### PR DESCRIPTION
**Pull request by souissimai**

### CODE REVIEW

File: src/components/menus/directory-tree-contextual-menu.js [hunk 1]
```diff
@@ -178,6 +329,16 @@ const DirectoryTreeContextualMenu = (props) => {
...
                     },
+                    {
+                        messageDescriptorId: 'paste',
+                        callback: () => {
+                            pasteElement(
+                                directory?.elementUuid,
+                                selectionForCopy
+                            );
+                        },
+                        icon: <ContentPasteIcon fontSize="small" />,
```
Comment (Meklo) [inline src/components/menus/directory-tree-contextual-menu.js]: I reckon pasting should be eligible on public folders owned by other users, the features under `isAllowed()` are only available for items in one own folders

File: src/components/dialogs/contingency-list/edition/criteria-based/criteria-based-edition-dialog.js [hunk 2]
```diff
@@ -23,13 +23,20 @@ import { getCriteriaBasedSchema } from 'components/dialogs/commons/criteria-base
...
     [EQUIPMENT_TYPE]: yup.string().required(),
     ...getCriteriaBasedSchema(),
 });
-
+const noSelectionForCopy = {
+    sourceCaseUuid: null,
+    name: null,
+    description: null,
+    parentDirectoryUuid: null,
+};
```
Comment (Meklo) [inline src/components/dialogs/contingency-list/edition/criteria-based/criteria-based-edition-dialog.js]: Maybe declare the default selection structure only once as a constant and export it to all needed places ?
Is there a reason for it being different than the one declared initially in the redux store :
        ```sourceItemUuid: null,
        typeItem: null,
        nameItem: null,
        descriptionItem: null,
        parentDirectoryUuid: null,``` ?

File: src/components/menus/directory-tree-contextual-menu.js (hunk 1 above)
Comment (souissimai) [reply src/components/menus/directory-tree-contextual-menu.js]: yes will fix it

File: src/components/dialogs/contingency-list/edition/criteria-based/criteria-based-edition-dialog.js (hunk 2 above)
Comment (souissimai) [reply src/components/dialogs/contingency-list/edition/criteria-based/criteria-based-edition-dialog.js]: The one that is created in the redux store is the intial state of **selectionForCopy**, that can be changed after copying element. The Other one is a const **noSelectionForCopy** to clean clipboard and then to clean the selectionForCopy but I will create it somewhere and export it as constant.

File: src/components/menus/directory-tree-contextual-menu.js [hunk 3]
```diff
@@ -181,7 +332,16 @@ const DirectoryTreeContextualMenu = (props) => {
...
                 );
             }
-
+            menuItems.push(
+                {
+                    messageDescriptorId: 'paste',
+                    callback: () => {
+                        pasteElement(directory?.elementUuid, selectionForCopy);
+                    },
+                    icon: <ContentPasteIcon fontSize="small" />,
```
Comment (Meklo) [inline src/components/menus/directory-tree-contextual-menu.js:341]: Maybe disable the option if the selectionForCopy item is null ?

File: src/components/dialogs/contingency-list/edition/criteria-based/criteria-based-edition-dialog.js [hunk 4]
```diff
@@ -95,6 +100,13 @@ const CriteriaBasedEditionDialog = ({
     const onSubmit = (contingencyList) => {
         editContingencyList(contingencyListId, contingencyList)
             .then(() => {
+                if (selectionForCopy.sourceItemUuid === contingencyListId) {
+                    dispatch(setSelectionForCopy(noSelectionForCopy));
+                    broadcastChannel.postMessage({
+                        noSelectionForCopy,
+                    });
+                }
+                dispatch(setSelectionForCopy(noSelectionForCopy));
```
Comment (AbdelHedhili) [inline src/components/dialogs/contingency-list/edition/criteria-based/criteria-based-edition-dialog.js]: what is this dispatch for ? and why is it only present here and not in `explicit-nameing-edition-dialog.js` and `script-dialog-edition`

**Comment by AbdelHedhili** [comment]
There is a bug : 

- open 2 browser (one in private mode) and connect each to a different user
- copy something with user A
- paste it with user B
- snackbar error

**Review by AbdelHedhili — CHANGES_REQUESTED** [review]
test not ok

**Comment by souissimai** [comment]
> There is a bug :
> 
>     * open 2 browser (one in private mode) and connect each to a different user
> 
>     * copy something with user A
> 
>     * paste it with user B
> 
>     * snackbar error

Same Behavior detected for nodes (gridStudy)
The Broadcast Channel API is indeed a handy tool for communication between different browsing contexts or tabs within the same browser. However, in the case of a private window, it does nor work. Private browsing is designed to isolate each session from others to enhance privacy, which means communication between a regular window and a private window (or between two private windows) is typically restricted.

File: src/components/dialogs/contingency-list/edition/criteria-based/criteria-based-edition-dialog.js (hunk 4 above)
Comment (souissimai) [reply src/components/dialogs/contingency-list/edition/criteria-based/criteria-based-edition-dialog.js]: It was added  by mistake

**Review by Meklo — APPROVED** [review]

**Comment by AbdelHedhili** [comment]
> > There is a bug :
> > ```
> > * open 2 browser (one in private mode) and connect each to a different user
> > 
> > * copy something with user A
> > 
> > * paste it with user B
> > 
> > * snackbar error
> > ```
> 
> Same Behavior detected for nodes (gridStudy) The Broadcast Channel API is indeed a handy tool for communication between different browsing contexts or tabs within the same browser. However, in the case of a private window, it does nor work. Private browsing is designed to isolate each session from others to enhance privacy, which means communication between a regular window and a private window (or between two private windows) is typically restricted.

Indeed, will be handled later (if handled at all)

**Review by AbdelHedhili — APPROVED** [review]

**Pull request merged**